### PR TITLE
[FieldSensitivePL] Fix vectorization.

### PR DESCRIPTION
**Pull request by nate-chandler**
FieldSensitivePrunedLiveness is used as a vectorization of PrunedLiveness.  An instance of FSPL with N elements needs to be able to represent the same states as N instances of PL.

Previously, it failed to do that in two significant ways:

(1) It attempted to save space for which elements were live by using a range.  This failed to account for instructions which are users of non-contiguous fields of an aggregate.

    apply(
      @owned (struct_element_addr %s, #S.f1),
      @owned (struct_element_addr %s, #S.f3)
    )

(2) It used a single bit to represent whether the instruction was consuming.  This failed to account for instructions which consumed some fields and borrowed others.

    apply(
      @owned (struct_element_addr %s, #S.f1),
      @guaranteed (struct_element_addr %s, #S.f2)
    )

The fix for (1) is to use a bit vector to represent which elements are used by the instruction.  The fix for (2) is to use a second bit vector to represent which elements are _consumed_ by the instruction.

rdar://110909290

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

**Review by atrick — APPROVED** [review]
Awesome! I reviewed this for CCC.

File: include/swift/SIL/FieldSensitivePrunedLiveness.h [hunk 1]
```diff
@@ -848,19 +862,37 @@ class FieldSensitivePrunedLiveness {
...
+  }
+
+  /// Whether \p user uses the fields in \p range as indicated by \p kind.
+  bool isInterestingUserOfKind(SILInstruction *user, IsInterestingUser kind,
+                               TypeTreeLeafTypeRange range) const {
+    auto *record = getInterestingUser(user);
+    if (!record)
+      return kind == IsInterestingUser::NonUser;
+
+    for (auto element : range.getRange()) {
```
[the remark this answers is not in the document]
Comment (nate-chandler) [reply include/swift/SIL/FieldSensitivePrunedLiveness.h:891]: Folded a fix into https://github.com/apple/swift/pull/66728 via https://github.com/apple/swift/compare/c322ba0353dc6ddda92c9aaccd24a669b991b286..82ba69e31909925436d48b19bc40dc71f430f50c .

**Pull request merged**